core/Inst.hpp

```diff
@@ -44,6 +44,9 @@ namespace olympia
             void setOriginalDestination(const Reg & destination){
                 original_dest_ = destination;
             }
+            void setAddrReg(const Reg & addr_reg){
+                addr_reg_ = addr_reg;
+            }
```

Comment:
Did you mean data register?  There is only 1 data register and multiple source registers for address.  I was thinking that the `src_` register list would be for address operands and `addr_reg_` would actually be `data_reg_`

Reply:
Not sure if I'm following of how we would have multiple source registers for address? For stores we would only ever have 1 data operand and 1 address operand?

From a coding standpoint I get why we should make src_ register the place for address operands because we can then always check src_registers for loads/stores like you explained in the other comment.

Reply:
Ugh, I keep mixing ARM/PowerPC ISA with RISC-V.  Indeed, there are no load/store instructions that use 2 integer sources for address calculation, just 1.

So, let me start over with my thinking reset.  🤣 

Yes, I think the `src_` register list should be only for address operands (sounds like you agree) and we should have a `data_src_reg_` for stores.  

> Also because we're using a separate mask for tracking address operands, do we then allocate a PRF for that address operand? 

Yes, you will always _refer_ to a PRF for the address sources just like any other register.  And yes, you will refer to a PRF for the store data just like any other register.  Those values usually come from an integer/float instruction like an `addi` or from a load (think memcpy).

In modern processors, MBRs are usually called something like store buffers or line fill buffers.  Either way, the memory to be stored comes from a register in the PRF.

Reply:
Addressed in [453b6c3](https://github.com/riscv-software-src/riscv-perf-model/pull/70/commits/453b6c3b19892012263c7985f4af0fc4717b05d1)